RevenueCatUI/Templates/Components/Packages/PurchaseButton/PurchaseButtonComponentView.swift

```diff
@@ -0,0 +1,142 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchaseButtonComponentView.swift
+//
+//  Created by Josh Holtz on 9/27/24.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct PurchaseButtonComponentView: View {
+
+    private let viewModel: PurchaseButtonComponentViewModel
+
+    internal init(viewModel: PurchaseButtonComponentViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        Button {
+            // WIP: Need to perform purchase logic
```

Comment:
I know of a pretty cool open-source SDK that can help out with this 😉

Reply:
I think I'll just use StoreKit 2 manually 😛